### PR TITLE
added updaterecommendation API details into MD file

### PR DESCRIPTION
**Pull request by msvinaykumar**

### CODE REVIEW

File: design/MonitoringModeAPI.md [hunk 1]
```diff
@@ -1997,3 +2015,166 @@ Returns the recommendation at a particular timestamp if it exists
...
+## Update Recommendations API
+
+Generate the recommendations for a specific experiment based on provided parameters.
+
+**Request Parameters**
+
+| Parameter         | Type   | Required | Description                                                           |
+|-------------------|--------|----------|-----------------------------------------------------------------------|
+| experiment_name   | string | Yes      | The name of the experiment.                                           |
+| intervalEndTime   | string | Yes      | The end time of the interval in the format "yyyy-MM-ddTHH:mm:sssZ".   |
```
Comment (chandrams) [inline design/MonitoringModeAPI.md:2028]: @msvinaykumar - Should intervalEndTime and intervalStartTime be interval_end_time and interval_start_time?

File: design/MonitoringModeAPI.md [hunk 2]
```diff
@@ -70,14 +70,17 @@ see [Update results](/design/UpdateResults.md)
 
 **NOTE:** The update to results of a particular experiment should follow Time Series Order to get valid recommendations.
```
Comment (chandrams) [inline design/MonitoringModeAPI.md:71]: We will need to remove references to Time series order or chronological order as we are now supporting non-chronological order. Please update the documentation

File: design/MonitoringModeAPI.md [hunk 3]
```diff
@@ -1997,3 +2015,166 @@ Returns the recommendation at a particular timestamp if it exists
...
+                        {
+                          "type": "info",
+                          "message": "There is not enough data available to generate a recommendation."
+                        }
+                      ]
+                    },
+                    "long_term": {
+                      "pods_count": 0,
+                      "confidence_level": 0.0,
+                      "notifications": [
```
Comment (chandrams) [inline design/MonitoringModeAPI.md:2149]: Notifications is in older format, please update with the latest json

File: design/MonitoringModeAPI.md [hunk 4]
```diff
@@ -1997,3 +2015,166 @@ Returns the recommendation at a particular timestamp if it exists
...
+
+**Request**
+
+`POST /updateRecommendations?experiment_name=?&intervalEndTime=?`
+
+`POST /updateRecommendations?experiment_name=?&intervalEndTime=?&intervalStartTime=?`
+
+example
+
+`curl --location --request POST 'http://127.0.0.1:8080/updateRecommendations?interval_end_time=2023-01-02T00:15:00.000Z&experiment_name=temp_1'`
```
[the remark this answers is not in the document]
Comment (msvinaykumar) [reply design/MonitoringModeAPI.md:2047]: done

File: design/MonitoringModeAPI.md (hunk 2 above)
Comment (msvinaykumar) [reply design/MonitoringModeAPI.md]: done

File: design/MonitoringModeAPI.md (hunk 3 above)
Comment (msvinaykumar) [reply design/MonitoringModeAPI.md]: done

File: design/MonitoringModeAPI.md [hunk 5]
```diff
@@ -1997,3 +2010,170 @@ Returns the recommendation at a particular timestamp if it exists
...
+
+**Error Responses**
+
+| HTTP Status Code | Description                                                                                        |
+|------------------|----------------------------------------------------------------------------------------------------|
+| 400              | experiment_name is mandatory.                                                                      |
+| 400              | interval_end_time is mandatory.                                                                    |
+| 400              | Given timestamp - \" 2023-011-02T00:00:00.000Z \" is not a valid timestamp format.                 |
+| 400              | Data not found!.                                                                                   |
+| 400              | The gap between the interval_start_time and interval_end_time must be within a maximum of 15 days! |
```
[the remark this answers is not in the document]
Comment (msvinaykumar) [reply design/MonitoringModeAPI.md:2177]: done

**Review by chandrams — APPROVED** [review]
LGTM

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**